src/App.tsx

```diff
@@ -15,71 +16,101 @@ export const goods = [
-      Jam is selected
+function getReorderedGoods(
+  goods: string[],
+  sortType: SortType,
+  isReversed: boolean,
+) {
+  const visibleGoods = [...goods];
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  visibleGoods.sort((g1, g2) => {
+    switch (sortType) {
+      case SortType.LENGTH:
+        return g1.length - g2.length;
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+      case SortType.ALPHABET:
+        return g1.localeCompare(g2);
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+      default:
+        return 0;
+    }
+  });
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
+export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+  const goods = getReorderedGoods(goodsFromServer, sortType, isReversed);
+  const resetButton = () => {
+    setSortType(SortType.NONE);
+    setIsReversed(false);
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          onClick={() => setSortType(SortType.ALPHABET)}
+          className={cn('button is-info', {
+            'is-light': sortType !== SortType.ALPHABET,
+          })}
+        >
+          Sort alphabetically
+        </button>
 
-        <tr data-cy="Good">
-          <td>
+        <button
+          type="button"
+          onClick={() => setSortType(SortType.LENGTH)}
+          className={cn('button is-success', {
+            'is-light': sortType !== SortType.LENGTH,
+          })}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => setIsReversed(prevIsReversed => !prevIsReversed)}
+        >
+          Reverse
+        </button>
+
+        {(sortType || isReversed)
+          && (
             <button
-              data-cy="AddButton"
               type="button"
-              className="button"
+              onClick={() => resetButton()}
```

Comment:
You can simplify this syntax. And would be great to rename the `resetButton` function to show that it's an action 
```suggestion
              onClick={onReset}
```